examples/ola-rdm.cpp

```diff
@@ -207,16 +207,22 @@ void DisplayHelpAndExit(const options &opts) {
   exit(ola::EXIT_USAGE);
 }
 
-
 /*
  * Dump the list of known pids
  */
 void DisplayPIDsAndExit(uint16_t manufacturer_id,
-                        const PidStoreHelper &pid_helper) {
+                        const PidStoreHelper &pid_helper,
+                        bool set_mode) {
+  vector<const ola::rdm::PidDescriptor *> pids;
+  pid_helper.SupportedPids(manufacturer_id, &pids);
+  // Remove PIDs that don't support the proper command class (GET with ola_rdm_get, SET with ola_rdm_set)
   vector<string> pid_names;
-  pid_helper.SupportedPids(manufacturer_id, &pid_names);
-  sort(pid_names.begin(), pid_names.end());
-
+  vector<const ola::rdm::PidDescriptor *>::const_iterator it_pids = pids.begin();
+  for (; it_pids != pids.end(); ++it_pids) {
+    if ((set_mode && (*it_pids)->SetRequest() != NULL) || (!set_mode && (*it_pids)->GetRequest() != NULL)) {
+      pid_names.push_back((*it_pids)->Name());
```

Comment:
I.e. I think this and the cout stuff can be one operation if they are sorted on the way into this for loop.